src/interceptors/add_grant_type_to_params.js

```diff
@@ -0,0 +1,23 @@
+/**
+   See what credentials (`username`, `password`, and `authorizationCode`) are
+   passed in params and set the grant_type based on those.
+
+   Changes to `ctx`:
+
+   - add `params.grant_type`
+ */
+export default class AddGrantTypeToParams {
+  enter({ params, ...ctx }) {
+    const { username, password, code } = params;
```

Comment:
Maybe change param name `code` to `authorizationCode` so that it would be obvious we want to check the authorization code.

Reply:
True it could be more descriptive. However, all the body params in SDK methods pass them to the API requests as they are, and the OAuth2 authorization code token request requires a `code` param so therefore I'd keep it as it is.